packages/smarthr-ui/src/libs/debounce.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * 関数の呼び出しを指定された待機時間 (wait) だけ遅延させるデバウンス関数。
+ * 直前に同じ関数が呼び出された場合、タイマーをリセットして再び遅延させます。
+ *
+ * @param {T} func - 実行する関数
+ * @param {number} wait - デバウンスの待機時間（ミリ秒）
+ * @returns {(...args: Parameters<T>) => void} デバウンスされた関数
+ *
+ * @example
+ * const debouncedFunction = debounce(() => console.log('Called!'), 200);
+ */
+export const debounce = <T extends (...args: any[]) => void>(
+  func: T,
+  wait: number,
+): ((...args: Parameters<T>) => void) => {
+  let timeoutId: NodeJS.Timeout | string | number | null = null
```

Comment:
clearTimeoutの引数の型定義を参考にした
```
        /**
         * Cancels a `Timeout` object created by `setTimeout()`.
         * @since v0.0.1
         * @param timeout A `Timeout` object as returned by {@link setTimeout} or the `primitive` of the `Timeout` object as a string or a number.
         */
```

Reply:
※ブラウザ環境での実行時にはsetTimeoutはnumberを返す

Reply:
`ReturnType<typeof setTimeout>` でブラウザ環境もNode環境もよしなに型判定してくれるかも？